Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java

```diff
@@ -84,6 +90,19 @@
 
         if (entry != null) {
 
+            try {
+                // Check if the id used in a generic id. If so fetch the concept
+                // wrapper id for the generic wordnet id
+                if (conceptTypesService
+                        .getConceptTypeByConceptId(pathParts[lastIndex]) == ConceptTypes.GENERIC_WORDNET_CONCEPT) {
+                    ConceptEntry wrapperConceptEntry = conceptManager.getConceptEntry(entry.getId());
+                    entry.setId(wrapperConceptEntry.getId());
```

Comment:
I have added few details from wrappedConceptEntry to entry. Please let me know if I need add any other fields.